svm/src/account_loader.rs

```diff
@@ -2227,4 +2371,271 @@ mod tests {
 
         assert_eq!(actual_inspected_accounts, expected_inspected_accounts,);
     }
+
+    #[test]
+    fn test_load_transaction_accounts_data_sizes() {
+        let mut mock_bank = TestCallbacks::default();
+
+        let mut next_size = 1;
+        let mut mk_account = |pubkey, owner, executable| {
+            let size = next_size;
+            let account = AccountSharedData::create(
+                LAMPORTS_PER_SOL,
+                vec![0; size],
+                owner,
+                executable,
+                u64::MAX,
+            );
+
+            mock_bank.accounts_map.insert(pubkey, account.clone());
+
+            // accounts are counted at most twice
+            // by multiplying account size by 4, we ensure all totals are unique
+            next_size *= 4;
+
+            (size as u32, account)
+        };
+
+        let (native_loader_size, _) = mk_account(native_loader::id(), native_loader::id(), true);
+        let (bpf_loader_size, _) = mk_account(bpf_loader::id(), native_loader::id(), true);
+        let (upgradeable_loader_size, _) =
+            mk_account(bpf_loader_upgradeable::id(), native_loader::id(), true);
+
+        let program1_keypair = Keypair::new();
+        let program1 = program1_keypair.pubkey();
+        let (program1_size, _) = mk_account(program1, bpf_loader::id(), true);
+
+        let program2_keypair = Keypair::new();
+        let program2 = program2_keypair.pubkey();
+        let (program2_size, _) = mk_account(program2, bpf_loader_upgradeable::id(), true);
+
+        let programdata2_keypair = Keypair::new();
+        let programdata2 = programdata2_keypair.pubkey();
```

Comment:
nit: don't need keypairs for anything except the fee_payer. just use `Pubkey::new_unique` for the program ids.